java/src/jmri/InstanceManager.java

```diff
@@ -771,6 +771,45 @@ public InstanceManager() {
         }
     }
 
+    /**
+     * Get a list of all named beans from the managers.
+     *
+     * @return a list of all named beans from the managers or an empty list
+     */
+    @SuppressWarnings("unchecked") // the cast here is protected by the structure of the managerLists
+    @Nonnull
+    public List<Object> getAllNamedBeansFromManagers() {
+        log.trace("Get list of all instances");
+        
+        List<Object> list = new ArrayList<>();
+        for (Class<?> type : managerLists.keySet()) {
+            synchronized (type) {
+                for (Object manager : managerLists.get(type)) {
+                    if (manager instanceof jmri.Manager) {
```

Comment:
I don't quite understand the `if` structure here.  If you're looking for jmri.Managers (which manage NamedBeans), why are the "Ignore this" clauses needed?

Reply:
I'm a bit in the dark here, so I added the `if` statement to ensure that I don't miss something. The important part is the `throw` statement at the bottom. If a new manager type is added later to the instance manager but this method doesn't know about it, the method will throw an exception.

But if it is absolute sure that there will never be a manager that isn't an instance of `jmri.Manager` that holds any bean that we are interested in, then that `if` statement can be removed. It is probably so, and if you agree with that I will remove the `if` statement. But I want to be sure before I remove it.